### ISSUE TITLE
E731 not raised when assigned via tuple unpacking

### ISSUE THREAD

**Issue by jpy-git** [PyCQA/pycodestyle#1061]
```python
a = lambda x: x  # Raises E731
a, b = lambda x: x, lambda y: y  # Doesn't raise E731
a, b = 1, lambda y: y  # Doesn't raise E731
```

**Comment by Sam-Cleveland** [comment]
I think the best way to resolve this is to just mark all `logical_line`s with an '=' that has a 'lambda' at some point after it as having a lambda assignment. This covers all cases here and I can't think of any additional unwanted cases that this would include (as long as it is not an assignment operator or function declaration annotation, but these already checked and excluded). I'll make a PR implementing this.

**Comment by FichteFoll** [comment]
> I think the best way to resolve this is to just mark all `logical_line`s with an '=' that has a 'lambda' at some point after it as having a lambda assignment.

```py
mapped = map(lambda i: i * 2, range(10))
```
would also be matched by this when it shouldn't, if I understood you correctly?

**Comment by Sam-Cleveland** [comment]
Oh, you're right it is not as simple as what I had said.
> ```python
> mapped = map(lambda i: i * 2, range(10))
> ```

In addition to this, 
```python
f.method = lambda: 'Method'
```
and
```python
f['a'] = lambda x: x ** 2
```
both should be allowed according to tests/E73.py
Nonetheless, I came up with a regex that should only search for actual lambda assignments (making sure to account for all of the examples above). Additionally, regex is a more elegant solution than the way it was already being implemented. Making a PR.
